docs/Account/methods.md

```diff
@@ -0,0 +1,284 @@
+
+const blogContract = new ethers.Contract(
+  contractAddress,
+  abi, // contract abi
+  provider,
+);
+
+const createComment =
+  await blogContract.populateTransaction.addComment("comment");
+
+const tx1 = {
+  to: contractAddress,
+  data: createComment.data,
+};
+
+const userOp = await smartAccount.buildUserOp([tx1]);
+```
+
+**Parameters**
+- transactions (`Transaction[]`, required): The required argument  is an array of transactions. You can pass multiple transactions into a userOp if you would like to batch them together into one transaction.
+- buildUseropDto (`BuildUserOpOptions`): One can also pass these options to customize how a userOp is built.
+
+  ```ts
+  type BuildUserOpOptions = {
+    overrides?: Overrides;
+    skipBundlerGasEstimation?: boolean;
+    params?: ModuleInfo;
+    nonceOptions?: NonceOptions; 
+    forceEncodeForBatch?: boolean;
+    paymasterServiceData?: SponsorUserOperationDto;
+  }
+  ```
+  Let's look at each of these params:
+  1. overrides (`Overrides`): one can override any of the values of the userOp when it is being constructed.
+
+      ```ts
+      type BigNumberish = BigNumber | Bytes | bigint | string | number;
+
+      type Overrides = {
+        callGasLimit?: BigNumberish;
+        verificationGasLimit?: BigNumberish;
+        preVerificationGas?: BigNumberish;
+        maxFeePerGas?: BigNumberish;
+        maxPriorityFeePerGas?: BigNumberish;
+        paymasterData?: string;
+        signature?: string; // todo - add the definition
+      }
+      ```
+
+  2. skipBundlerGasEstimation (`boolean`): This boolean param can be set to false to skip some Gas Estimations on the bundler which can help with reducing latency on requests.
+
+  3. params (`ModuleInfo`): One can use this param to pass session validation module parameters. Refer to the tutorial to learn more about the session keys.
+
+      ```ts
+      type ModuleInfo = {
+        sessionID?: string;
+        sessionSigner?: Signer;
+        sessionValidationModule?: string;
+        additionalSessionData?: string;
+        batchSessionParams?: SessionParams[];
+      }
+      ```
+  4. nonceOptions(`NonceOptions`) : this can be used to execute multiple transactions in parallel. One can implement their own logic for nonce management.
+
+      ```ts
+      type NonceOptions = {
+        nonceKey?: number;
+        nonceOverride?: number;
+      };
+      // nonceOptions usage
+      let i = 0;
+      const userOp = await smartAccount.buildUserOp([tx1], {
+        nonceOptions: { nonceKey: i++ },
+      });
+      ```
+      Nonce can be initialised at any arbitrary number and incremented as one builds transactions. The nonceKey will create a batch or space in which the nonce can safely increment without colliding with other transactions. The nonceOverride will directly override the nonce and should only be if you know the order in which you are sending the userOps.
+
+    
+    
+  5. forceEncodeForBatch (`boolean`): 
```

Comment:
When you provide array of transactions, by default Biconomy sdk encode it for executeBatch() executor function and execute() function for single transaction. But in some cases you might want to encode single transaction for a batch if your custom module decodes only for executeBatch. 
You can provide this flag to true in such case. default behaviour: false